bee-api/bee-rest-api/src/endpoints/routes/api/v2/output.rs

```diff
@@ -1,87 +1,63 @@
-        .and(with_storage(storage))
-        .and(with_consensus_worker(consensus_worker))
-        .and_then(
-            |output_id, storage, consensus_worker| async move { output(output_id, storage, consensus_worker).await },
-        )
-        .boxed()
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/outputs/:output_id", get(output::<B>))
 }
 
 pub(crate) async fn output<B: StorageBackend>(
-    output_id: OutputId,
-    storage: ResourceHandle<B>,
-    consensus_worker: mpsc::UnboundedSender<ConsensusWorkerCommand>,
-) -> Result<impl Reply, Rejection> {
+    Path(output_id): Path<OutputId>,
+    Extension(args): Extension<ApiArgsFullNode<B>>,
+) -> Result<impl IntoResponse, ApiError> {
     let (cmd_tx, cmd_rx) = oneshot::channel::<(Result<Option<CreatedOutput>, Error>, LedgerIndex)>();
 
-    if let Err(e) = consensus_worker.send(ConsensusWorkerCommand::FetchOutput(output_id, cmd_tx)) {
-        error!("request to consensus worker failed: {}.", e);
+    if let Err(e) = args
+        .consensus_worker
+        .send(ConsensusWorkerCommand::FetchOutput(output_id, cmd_tx))
+    {
+        error!("request to consensus worker failed: {}", e);
+        return Err(ApiError::InternalError);
     }
 
-    match cmd_rx.await.map_err(|e| {
-        error!("response from consensus worker failed: {}.", e);
-        reject::custom(CustomRejection::ServiceUnavailable(
-            "unable to fetch the output".to_string(),
-        ))
-    })? {
+    let consensus_worker_response = cmd_rx.await.map_err(|e| {
+        error!("response from consensus worker failed: {}", e);
+        ApiError::InternalError
+    })?;
+
+    match consensus_worker_response {
         (Ok(response), ledger_index) => match response {
             Some(output) => {
-                let is_spent = Fetch::<OutputId, ConsumedOutput>::fetch(&*storage, &output_id).map_err(|e| {
-                    error!("unable to fetch the output: {}", e);
-                    reject::custom(CustomRejection::ServiceUnavailable(
-                        "unable to fetch the output".to_string(),
-                    ))
-                })?;
+                let consumed_output =
+                    Fetch::<OutputId, ConsumedOutput>::fetch(&*args.storage, &output_id).map_err(|e| {
+                        error!("cannot fetch from storage: {}", e);
+                        ApiError::InternalError
+                    })?;
 
-                Ok(warp::reply::json(&OutputResponse {
+                Ok(Json(OutputResponse {
                     message_id: output.message_id().to_string(),
                     transaction_id: output_id.transaction_id().to_string(),
                     output_index: output_id.index(),
-                    is_spent: is_spent.is_some(),
+                    is_spent: consumed_output.is_some(),
                     // TODO
                     milestone_index_spent: None,
                     // TODO
```

Comment:
I know it's not introduced in this PR, but can we document or remove or fix those TODOs?

Reply:
done